diff --git a/dist/directive-types.d.ts b/dist/directive-types.d.ts
index 82be2fca38ee0e061427d3f4e3db6e8dfef7b7c6..67ed6a5e25b42de8c6e6177ce98f7fffe466a10e 100644
--- a/dist/directive-types.d.ts
+++ b/dist/directive-types.d.ts
@@ -1,5 +1,5 @@
 import type { BlockContent, Data as MdastData, DefinitionContent, Parent as MdastParent, PhrasingContent } from "mdast";
-export type DirectiveAttributes = Record<string, string>;
+export type DirectiveAttributes = Record<string, string | null | undefined>;
 export interface ContainerDirective extends MdastParent {
     type: "containerDirective";
     name: string;
