diff --git a/index.js b/index.js
index 2724bd0b5c742042af20ec289ac30c6bf9be793c..0bb9980ba6d3bff49451742e9a20f72c266e2041 100644
--- a/index.js
+++ b/index.js
@@ -8162,9 +8162,7 @@ function loadTsconfig(config, svelteMap) {
         options: {
             ...options,
             noEmit: false, // Set to true in case of jsconfig, force false, else nothing is emitted
-            moduleResolution: 
-            // NodeJS: up to 4.9, Node10: since 5.0
-            (_a = ts.ModuleResolutionKind.NodeJs) !== null && _a !== void 0 ? _a : ts.ModuleResolutionKind.Node10, // Classic if not set, which gives wrong results
+            moduleResolution: ts.ModuleResolutionKind.NodeNext,
             declaration: true, // Needed for d.ts file generation
             emitDeclarationOnly: true, // We only want d.ts file generation
             declarationDir: config.declarationDir, // Where to put the declarations
diff --git a/index.mjs b/index.mjs
index a5ec3af3cb1f755d5c9c8de9528c9dd8cdcc5c91..65f2732e73e2988598aeb226c202cab5d11bb124 100644
--- a/index.mjs
+++ b/index.mjs
@@ -8142,9 +8142,7 @@ function loadTsconfig(config, svelteMap) {
         options: {
             ...options,
             noEmit: false, // Set to true in case of jsconfig, force false, else nothing is emitted
-            moduleResolution: 
-            // NodeJS: up to 4.9, Node10: since 5.0
-            (_a = ts.ModuleResolutionKind.NodeJs) !== null && _a !== void 0 ? _a : ts.ModuleResolutionKind.Node10, // Classic if not set, which gives wrong results
+            moduleResolution: ts.ModuleResolutionKind.NodeNext,
             declaration: true, // Needed for d.ts file generation
             emitDeclarationOnly: true, // We only want d.ts file generation
             declarationDir: config.declarationDir, // Where to put the declarations
