diff --git a/index.js b/index.js
index b36a9debef01c910720fd5c678debc961126a1fc..bcd2c47e1cfd710c60743beda6db5717b84a4fc4 100644
--- a/index.js
+++ b/index.js
@@ -8088,9 +8088,7 @@ function loadTsconfig(config, svelteMap) {
         options: {
             ...options,
             noEmit: false, // Set to true in case of jsconfig, force false, else nothing is emitted
-            moduleResolution: 
-            // NodeJS: up to 4.9, Node10: since 5.0
-            (_a = ts.ModuleResolutionKind.NodeJs) !== null && _a !== void 0 ? _a : ts.ModuleResolutionKind.Node10, // Classic if not set, which gives wrong results
+            moduleResolution: ts.ModuleResolutionKind.NodeNext,
             declaration: true, // Needed for d.ts file generation
             emitDeclarationOnly: true, // We only want d.ts file generation
             declarationDir: config.declarationDir, // Where to put the declarations
diff --git a/index.mjs b/index.mjs
index e6d39d1bdeb8df30da84194834736e35e5cea741..e5b2ba29e7b4c83b876353e6cb171ba7191e4c78 100644
--- a/index.mjs
+++ b/index.mjs
@@ -8068,9 +8068,7 @@ function loadTsconfig(config, svelteMap) {
         options: {
             ...options,
             noEmit: false, // Set to true in case of jsconfig, force false, else nothing is emitted
-            moduleResolution: 
-            // NodeJS: up to 4.9, Node10: since 5.0
-            (_a = ts.ModuleResolutionKind.NodeJs) !== null && _a !== void 0 ? _a : ts.ModuleResolutionKind.Node10, // Classic if not set, which gives wrong results
+            moduleResolution: ts.ModuleResolutionKind.NodeNext,
             declaration: true, // Needed for d.ts file generation
             emitDeclarationOnly: true, // We only want d.ts file generation
             declarationDir: config.declarationDir, // Where to put the declarations
