diff --git a/index.js b/index.js
index 78ac20a6b0f3bbe5ae9242d379b8f4388b2c7bb4..eedbcbd516293e0fcfa4c62b1dd6479e6233090e 100644
--- a/index.js
+++ b/index.js
@@ -8022,9 +8022,7 @@ function loadTsconfig(config, svelteMap) {
         options: {
             ...options,
             noEmit: false, // Set to true in case of jsconfig, force false, else nothing is emitted
-            moduleResolution: 
-            // NodeJS: up to 4.9, Node10: since 5.0
-            (_a = ts.ModuleResolutionKind.NodeJs) !== null && _a !== void 0 ? _a : ts.ModuleResolutionKind.Node10, // Classic if not set, which gives wrong results
+            moduleResolution: ts.ModuleResolutionKind.NodeNext,
             declaration: true, // Needed for d.ts file generation
             emitDeclarationOnly: true, // We only want d.ts file generation
             declarationDir: config.declarationDir, // Where to put the declarations
diff --git a/index.mjs b/index.mjs
index dda691d65c194fa42b38f813e0ad20106c36bf76..8ee6cb8d1d6e03c2735a9138e9112f684e5dc2af 100644
--- a/index.mjs
+++ b/index.mjs
@@ -8002,9 +8002,7 @@ function loadTsconfig(config, svelteMap) {
         options: {
             ...options,
             noEmit: false, // Set to true in case of jsconfig, force false, else nothing is emitted
-            moduleResolution: 
-            // NodeJS: up to 4.9, Node10: since 5.0
-            (_a = ts.ModuleResolutionKind.NodeJs) !== null && _a !== void 0 ? _a : ts.ModuleResolutionKind.Node10, // Classic if not set, which gives wrong results
+            moduleResolution: ts.ModuleResolutionKind.NodeNext,
             declaration: true, // Needed for d.ts file generation
             emitDeclarationOnly: true, // We only want d.ts file generation
             declarationDir: config.declarationDir, // Where to put the declarations
