diff --git a/index.js b/index.js
index f1e2c2f37581f14a224fe1dbf8d06de01ac4c6da..4b485d9ee892bd2fdb33cc25f6870ea9c5864d73 100644
--- a/index.js
+++ b/index.js
@@ -8106,9 +8106,7 @@ function loadTsconfig(config, svelteMap) {
         options: {
             ...options,
             noEmit: false, // Set to true in case of jsconfig, force false, else nothing is emitted
-            moduleResolution: 
-            // NodeJS: up to 4.9, Node10: since 5.0
-            (_a = ts.ModuleResolutionKind.NodeJs) !== null && _a !== void 0 ? _a : ts.ModuleResolutionKind.Node10, // Classic if not set, which gives wrong results
+            moduleResolution: ts.ModuleResolutionKind.NodeNext,
             declaration: true, // Needed for d.ts file generation
             emitDeclarationOnly: true, // We only want d.ts file generation
             declarationDir: config.declarationDir, // Where to put the declarations
diff --git a/index.mjs b/index.mjs
index 60ddf438e34b2a0775924b3a9117e906a55159d5..9d3305758ee629572c172499df021dfa348d3c77 100644
--- a/index.mjs
+++ b/index.mjs
@@ -8086,9 +8086,7 @@ function loadTsconfig(config, svelteMap) {
         options: {
             ...options,
             noEmit: false, // Set to true in case of jsconfig, force false, else nothing is emitted
-            moduleResolution: 
-            // NodeJS: up to 4.9, Node10: since 5.0
-            (_a = ts.ModuleResolutionKind.NodeJs) !== null && _a !== void 0 ? _a : ts.ModuleResolutionKind.Node10, // Classic if not set, which gives wrong results
+            moduleResolution: ts.ModuleResolutionKind.NodeNext,
             declaration: true, // Needed for d.ts file generation
             emitDeclarationOnly: true, // We only want d.ts file generation
             declarationDir: config.declarationDir, // Where to put the declarations
