diff --git a/index.js b/index.js
index c999969cf4681dceb4078879126e01b16253b18a..b4f7e909ef30299b3bb7ba60d244c42f402ccf94 100644
--- a/index.js
+++ b/index.js
@@ -8125,9 +8125,7 @@ function loadTsconfig(config, svelteMap) {
         options: {
             ...options,
             noEmit: false, // Set to true in case of jsconfig, force false, else nothing is emitted
-            moduleResolution: 
-            // NodeJS: up to 4.9, Node10: since 5.0
-            (_a = ts.ModuleResolutionKind.NodeJs) !== null && _a !== void 0 ? _a : ts.ModuleResolutionKind.Node10, // Classic if not set, which gives wrong results
+            moduleResolution: ts.ModuleResolutionKind.NodeNext,
             declaration: true, // Needed for d.ts file generation
             emitDeclarationOnly: true, // We only want d.ts file generation
             declarationDir: config.declarationDir, // Where to put the declarations
diff --git a/index.mjs b/index.mjs
index e89e944bd198dd8487afd3851bc4d33bca30a1e7..f328c99ae2d222e366f6926bcfcf6e3e59f862a6 100644
--- a/index.mjs
+++ b/index.mjs
@@ -8105,9 +8105,7 @@ function loadTsconfig(config, svelteMap) {
         options: {
             ...options,
             noEmit: false, // Set to true in case of jsconfig, force false, else nothing is emitted
-            moduleResolution: 
-            // NodeJS: up to 4.9, Node10: since 5.0
-            (_a = ts.ModuleResolutionKind.NodeJs) !== null && _a !== void 0 ? _a : ts.ModuleResolutionKind.Node10, // Classic if not set, which gives wrong results
+            moduleResolution: ts.ModuleResolutionKind.NodeNext,
             declaration: true, // Needed for d.ts file generation
             emitDeclarationOnly: true, // We only want d.ts file generation
             declarationDir: config.declarationDir, // Where to put the declarations
